pages/wheelPage.js

```diff
@@ -0,0 +1,109 @@
+import React, { useState } from "react";
+import { useEffect } from "react";
+import { StyledImage } from "@/components/StyledImage";
+import styled from "styled-components";
+import { StyleBox } from "@/components/StyledBox";
+import { StyledLink } from "@/components/StyledLink";
+import Link from "next/link";
+import { activities } from "@/lib/users/activities/activities";
+
+const ImageContainer = styled.div`
+  position: relative;
+  height: 25rem;
+  margin: 0rem;
+  padding: 1rem;
+`;
+
+const PlayButton = styled.button`
+  display: block;
+  width: 80px;
+  height: 80px;
+  border: none;
+  background: lightcyan;
+  font-weight: bold;
+  font-size: large;
+  border-radius: 50%;
+  box-shadow: 0px 5px 20px rgba(0, 0, 0, 0.2);
+  cursor: pointer;
+  position: fixed;
+  bottom: calc(5%);
+  left: 50%;
+  transform: translateX(-50%);
+`;
+const BoxLink = styled(Link)`
+  text-decoration: none;
+  color: black;
+`;
+
+const FixedBackLink = styled(StyledLink)`
+  position: fixed;
+  bottom: 30px;
+  left: 30px;
+  box-shadow: 0px 5px 20px rgba(0, 0, 0, 0.2);
+`;
+
+function Wheel() {
+  const [selectedActivity, setSelectedActivity] = useState(null);
+  const [spin, setSpin] = useState(false);
+
+  useEffect(() => {
+    if (spin) {
+      const spinTimeout = setTimeout(() => {
+        setSelectedActivity(
+          activities[Math.floor(Math.random() * activities.length)]
+        );
+        setSpin(false);
+      }, 5000);
+
+      return () => clearTimeout(spinTimeout);
+    }
+  }, [spin]);
+
+  const handleSpinClick = () => {
+    setSelectedActivity(null);
+    setSpin(true);
+  };
+  const renderPrize = (activity, index) => {
+    const activityStyle = {
+      transform: `rotate(${(360 / activities.length) * index}deg)`,
+      backgroundColor: index % 2 === 0 ? "#C1E1C1" : "#87CEFA",
+    };
+
+    const spinWheel = () => {
+      const randomIndex = Math.floor(Math.random() * activities.length);
+      setSelectedActivity(activities[randomIndex]);
+    };
+  };
+  return (
+    <div>
+      <div>{activities.map((activity, id) => renderPrize(activity, id))}</div>
+
+      <ImageContainer>
+        <StyledImage
+          src="/wheel-of-fortune.png"
+          fill
+          sizes="(max-width: 768px) 100vw,
+              (max-width: 1200px) 50vw,
+              33vw"
+          alt=""
+        />
+      </ImageContainer>
+      <PlayButton onClick={handleSpinClick} disabled={spin}>
+        {spin ? "Spinning..." : "Spin the wheel!"}
+      </PlayButton>
+      <StyleBox>
+        {selectedActivity && (
+          <p>
+            Your next Adventure is: <p>{selectedActivity.name} </p>
+          </p>
+        )}
+        <BoxLink href={`/${selectedActivity?.id}`}>
+          {selectedActivity?.name}
+        </BoxLink>
+      </StyleBox>
```

Comment:
In your app it is very hard to see why the name is shown twice and that one of the times it is a Link. Maybe you can reconsider how this is shown.